ossdbs/api.py

```diff
@@ -94,6 +95,19 @@ def generate_electrodes(settings: dict):
                 rotation=rotation,
             )
 
+        # Apply hp-refinement only on active contacts
+        if "HPRefinement" in settings["Mesh"]:
+            hp_settings = settings["Mesh"]["HPRefinement"]
+            if hp_settings["Active"] and "Contacts" in electrode_parameters:
+                for contact_info in electrode_parameters["Contacts"]:
+                    if contact_info["Active"]:
+                        contact_idx = contact_info["Contact_ID"]
+                        order = hp_settings["Order"]
+                        _set_edge_hp_flag(electrode, {f"Contact_{contact_idx}": order})
```

Comment:
I would move this to the electrode class, this doesn't make sense in the API.
I.e, create a method and call it using `electrode.set_hp_flag` (should also be a public method)

Reply:
You are right; it is moved to the electrode_model_template.py as a public method and is called by ''electrode.set_hp_flag''